java/ql/lib/semmle/code/java/dataflow/internal/DataFlowUtil.qll

```diff
@@ -304,6 +304,27 @@ class ContentSet instanceof Content {
   }
 }
 
+/**
+ * Holds if `g` validates the `e` upon evaluating to `branch`.
+ *
+ * The expression `e` is expected to be a syntactic part of the guard `g`.
+ * For example, the guard `g` might be a call `isSafe(x)` and the expression `e`
+ * the argument `x`.
+ */
+signature predicate guardChecksSig(Guard g, Expr e, boolean branch);
+
+module BarrierGuard<guardChecksSig/3 guardChecks> {
```

Comment:
I think a QLDoc here explaining _how to use_ this parameterized module would be great! From _just_ reading the change in this file, I did not know whether I was expected to use this with `isBarrier`/`isBarrierIn`/`isBarrierOut`.

I think it could also still make sense to follow the [QLDoc style guide for modules](https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#qldoc-for-modules), such that it reads

```codeql
/**
 * Provides logic for barrier guards, where an expression is validated to be safe
 * in a specific branch.
 *
 * Expected to be used in dataflow/taint-tracking configurations with `isBarrier`/`isSanitizer`
 */
...
```

:thinking: initially I wrote the first line as `Provides logic for barrier guards` which feels like an autogenerated JavaDoc that doesn't bring much value...

Reply:
fixed